cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -375,11 +381,11 @@ impl LockedManifestCatalog {
 
         for (install_id, manifest_descriptor) in manifest.install.iter() {
             let resolved_descriptor = PackageDescriptor {
-                name: install_id.clone(),
-                pkg_path: manifest_descriptor.pkg_path.clone(),
+                install_id: install_id.clone(),
+                attr_path: manifest_descriptor.pkg_path.clone(),
                 derivation: None,
-                semver: None,
                 version: manifest_descriptor.version.clone(),
+                allow_pre_releases: manifest.options.semver.prefer_pre_releases,
```

Comment:
nit: "_allow_ pre-releases" and "_prefer_ pre-releases" have rather differnet meanings. Using these terms interchangibly will be quite confusing, esp. since pkgdb implemented the "_prefer_" part

Reply:
Agree on being careful about this.  I think it's currently implemented as _allow_.  Especially in resolution, there is no version sorting.  It's simply the latest page that meets ALL requirements.  If something has a pre-release version, that page is either allowed or not.